core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorBodyDirective.js

```diff
@@ -0,0 +1,128 @@
+            type: 'text',
+            value: newHtml
+          }]);
+          // We reload the initialState because a fresh copy of the state is
+          // created in the states dict on every update.
+          $scope.initialState = explorationStatesService.getState(
+            explorationInitStateNameService.savedMemento);
+        };
+
+        $scope.saveCustomizationArgs = function(
+            stateName, newCustomizationArgs) {
+          explorationStatesService.saveInteractionCustomizationArgs(
+            stateName, newCustomizationArgs);
+        };
+        $scope.saveAnswerGroups = function(stateName, newAnswerGroups) {
+          explorationStatesService.saveInteractionAnswerGroups(
+            stateName, newAnswerGroups);
+        };
+        $scope.saveDefaultOutcome = function(stateName, newDefaultOutcome) {
+          explorationStatesService.saveInteractionDefaultOutcome(
+            stateName, newDefaultOutcome);
+        };
+        $scope.saveBridgeText = function(stateName, newHtml) {
+          explorationStatesService.saveStateContent(stateName, [{
+            type: 'text',
+            value: newHtml
+          }]);
+          // We reload the questions because a fresh copy of the state is
+          // created in the states dict on every update.
+          // TODO(sll): Pass this back to the questions data service? But we
+          // need a way to update the questions here...
+          $scope.questions = (
+              SimpleEditorQuestionsDataService.getQuestionsInOrder());
+        };
+
+        // Requirement: for the last question in the list, there should be at
+        // least one answer group.
+        $scope.canAddNewQuestion = function() {
+          if ($scope.questions.length === 0) {
+            return true;
+          } else {
+            var lastQuestion = $scope.questions[$scope.questions.length - 1];
+            return lastQuestion.answerGroups.length > 0;
+          }
+        };
+
+        $scope.addNewQuestion = function() {
+          // Add a new multiple-choice interaction to the latest state, and
+          // re-calculate the list of questions.
+          var lastStateName = (
+            $scope.questions.length > 0 ?
+            $scope.questions[
+              $scope.questions.length - 1].answerGroups[0].outcome.dest :
+            explorationInitStateNameService.savedMemento);
+          explorationStatesService.saveInteractionId(
+            lastStateName, 'MultipleChoiceInput');
+          explorationStatesService.saveInteractionCustomizationArgs(
+            lastStateName, {
+              choices: {
+                value: ['']
+              }
+            }
+          );
+          $scope.questions = (
+              SimpleEditorQuestionsDataService.getQuestionsInOrder());
+          $rootScope.$broadcast('updateSimpleEditorSidebar');
+
+          // The new question needs to be loaded before this broadcast can have
```

Comment:
Actually, the timeout is still needed, because the new directive needs to initialize before we can send an event to it.